webdriver-spec.html

```diff
@@ -4282,36 +4282,50 @@ <h2 data-dfn-for="NavigatorAutomationInformation">Interface</h2>
 <p>An ECMAScript <a>Object</a> <dfn>represents a web element</dfn>
  if it has a <a>web element identifier</a> <a>own property</a>.
 
-<p>Each <a>browsing context</a> has an associated list of
- <dfn data-lt="known element">known elements</dfn>.
+<p>Each <a>browsing context</a> has an associated <dfn>list of
+ known elements</dfn>.
  When the <a>browsing context</a> is <a>discarded</a>,
- the list of <a>known elements</a> is discarded along with it.
-
-<p>To <dfn>get a known element</dfn>
- with argument <var>reference</var>,
- let <var>element</var> be the <a>known element</a>
- which <a>web element reference</a> matches <var>reference</var>.
- If there are none, return <a>error</a>
- with <a>error code</a> <a>no such element</a>.
- If <var>element</var> <a>is stale</a>, return
+ the <a>list of known elements</a> is discarded along with it.
+
+<p>To <dfn>get a known element</dfn> with
+ argument <var>reference</var>, run the following steps:
+
+<ol>
+ <li>Let <var>element</var> be the item in the <a>current browsing
+ context</a>'s <a>list of known elements</a> for which the <a>web
+ element reference</a> matches <var>reference</var>, if such an
+ element exists. Otherwise return <a>error</a> with <a>error
+ code</a> <a>no such element</a>.
+ <li>If <var>element</var> <a>is stale</a>, return
  <a>error</a> with <a>error code</a>
- <a>stale element reference</a>. Otherwise return
- <a>success</a> with <var>element</var>.
+ <a>stale element reference</a>.
+ <li>Return <a>success</a> with <var>element</var>.
+</ol>
+
+<p>To <dfn>get a known connected element</dfn> with
+argument <var>reference</var>, run the following steps:
+<ol>
+  <li>Let <var>element</var> be the result of <a>trying</a> to <a>get
+  a known element</a> with argument <var>reference</var>.
+  <li>If <var>element</var> is not <a>connected</a>
+  return <a>error</a> with error code <a>stale element reference</a>.
+  <li>Return <a>success</a> with <var>element</var>.
+</ol>
```

Comment:
@jgraham @shs96c A connected element doesn’t take into account that the document may have changed, see the definition of ‘is stale’.

WebDriver is special in this regard because element references are kept across documents and the DOM definition of ‘connected’ doesn’t take this into account.

Reply:
We should change the definition of "is stale" to just check if the node's document is the active document, or just inline that check.

Reply:
I added that earlier this week:

> An element is stale if its node document is not the active document or if its context object is not connected.

And step 2 of ‘get a known element’ says:

> If element is stale, return error with error code stale element reference. 

And the new ‘get a known connected element’ you added here calls that in step 1, so I think we are OK here! 👍 

Sorry for the false warning.

Reply:
The problem here is that "get a known element" is checking for connectedness, when that isn't necessary. It should be possible to do

```
elem = executeScript("window.e = document.createElement('div'); return e")
getElementTagName(e)
```

So stale should just check that the element exists, not that it's connected to the document.